.github/dev-containers/centralserver/files/cs-entrypoint.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+PACKAGED_VERSION="$(cat /root/VERSION)"
+INSTALLED_VERSION=$(dpkg-query --showformat='${Version}' --show xroad-center)
+
+log() { echo "$(date --utc -Iseconds) INFO [entrypoint] $*"; }
+warn() { echo "$(date --utc -Iseconds) WARN [entrypoint] $*" >&2; }
+
+wait_db() {
+  local count=0
+  while ((count++ < 60)) && ! pg_isready -q -t 2; do
+    sleep 1
+  done
+}
+
+log "Starting X-Road central server version $INSTALLED_VERSION"
+
+if [ "$INSTALLED_VERSION" == "$PACKAGED_VERSION" ]; then
+    if [ -f /etc/xroad/VERSION ]; then
+        CONFIG_VERSION="$(cat /etc/xroad/VERSION)"
+    else
+        warn "Current configuration version not known"
+        CONFIG_VERSION=
+    fi
+    if [ -n "$CONFIG_VERSION" ] && dpkg --compare-versions "$PACKAGED_VERSION" gt "$CONFIG_VERSION"; then
+        # Update X-Road configuration on startup, if necessary
+        log "Updating configuration from $CONFIG_VERSION to $PACKAGED_VERSION"
+        cp -a /root/etc/xroad/* /etc/xroad/
+        pg_ctlcluster 14 main start
+        wait_db
+        dpkg-reconfigure xroad-center
+        pg_ctlcluster 14 main stop
+        nginx -s stop
+        sleep 1
+        echo "$PACKAGED_VERSION" >/etc/xroad/version
+    fi
+    # Initialize TEST-CA if it has not been initialized yet and the directory exists
+    if [ -d /home/ca ] && [ ! -f /home/ca/CA/.init ]; then
+        log "Initializing TEST-CA..."
+        su ca -c 'cd /home/ca/CA && ./init.sh' >&/dev/null
+    fi
+else
+    echo "WARN: Installed version ($INSTALLED_VERSION) does not match packaged version ($PACKAGED_VERSION)" >&2
+fi
+
+if [  -n "$XROAD_TOKEN_PIN" ]
+then
+    log "XROAD_TOKEN_PIN variable set, writing to /etc/xroad/autologin"
+    echo "$XROAD_TOKEN_PIN" > /etc/xroad/autologin
+    unset XROAD_TOKEN_PIN
+fi
+
+if ! crudini --get /etc/xroad/conf.d/local.ini registration-service api-token &>/dev/null; then
+  log "Creating API token for registration service..."
```

Comment:
Consider mentioning the management service too since the same token is used for registration and management services.

Reply:
Updated.